scripts/Calibration/tofpd/group_calibration.py

```diff
@@ -0,0 +1,272 @@
+    :param previous_calibration: Optional previous diffcal workspace
+    :param output_basename: name to use for temporay and output workspace, default group_calibration
+    :param cc_kwargs: dict of parameters to pass to cc_calibrate_groups
+    :param pdcal_kwargs: dict of parameters to pass to pdcalibration_groups
+    :return: The final diffcal after running cc_calibrate_groups and  pdcalibration_groups
+    """
+
+    cc_diffcal = cc_calibrate_groups(data_ws,
+                                     group_ws,
+                                     output_basename,
+                                     previous_calibration,
+                                     **cc_kwargs)
+
+    diffcal, mask = pdcalibration_groups(data_ws,
+                                         group_ws,
+                                         cc_diffcal,
+                                         output_basename,
+                                         previous_calibration,
+                                         **pdcal_kwargs)
+
+    return diffcal, mask
+
+
+def process_json(json_filename):
+    """This will read a json file, process the data and save the calibration.
+
+    Only ``Calibrant`` and ``Groups`` are required.
+
+    An example input showing every possible options is:
+
+    .. code-block:: JSON
+
+      {
+        "Calibrant": "12345",
+        "Groups": "/path/to/groups.xml",
+        "Mask": "/path/to/mask.xml",
+        "Instrument": "NOM",
+        "Date" : "2019_09_04",
+        "SampleEnvironment": "shifter",
+        "PreviousCalibration": "/path/to/cal.h5",
+        "CalDirectory": "/path/to/output_directory",
+        "CrossCorrelate": {"Step": 0.001,
+                           "DReference: 1.5,
+                           "Xmin": 1.0,
+                           "Xmax": 3.0,
+                           "MaxDSpaceShift": 0.25},
+        "PDCalibration": {"PeakPositions": [1, 2, 3],
+                          "TofBinning": (300,0.001,16666),
+                          "PeakFunction": 'Gaussian',
+                          "PeakWindow": 0.1,
+                          "PeakWidthPercent": 0.001}
+      }
+    """
+    with open(json_filename) as json_file:
+        args = json.load(json_file)
+
+    calibrant = args['Calibrant']
+    groups = args['Groups']
+    sample_env = args.get('SampleEnvironment', 'UnknownSampleEnvironment')
+    mask = args.get('Mask')
+    instrument = args.get('Instrument', 'NOM')
+    cc_kwargs = args.get('CrossCorrelate', {})
+    pdcal_kwargs = args.get('PDCalibration', {})
+    previous_calibration = args.get('PreviousCalibration')
+
+    date = str(args.get('Date', datetime.datetime.now().strftime('%Y_%m_%d')))
+    caldirectory = str(args.get('CalDirectory', os.path.abspath('.')))
+
+    calfilename = f'{caldirectory}/{instrument}_{calibrant}_{date}_{sample_env}.h5'
+    print('going to create calibration file: %s' % calfilename)
```

Comment:
Change this to mantid's logging so it will be seen in the console interleaved with all the other messages